### PR TITLE
Bignum test

### LINKED ISSUE #78: Comparison doesn't work when one value is missing a decimal vector
For example, `(1.25 > 1)` should return `true`, but it is returning `false` in this case. Similarly, `(1 < 1.25)` should return `true`, but it returns `false`. 

So it seems to occur whenever the floors of the two values are the same, but one of them is an integer (i.e. equal to its own floor). Need to look at `libslide/src/bignum/compare.rs`

### PR DESCRIPTION
**Pull request by luke-bhan**
This adds tests and removes all leading and trailing zeros in both the int and dec vecs for all operations. It edits the constructor to fix #78, but does not handle the cmp changes required.

### CODE REVIEW

**Comment by luke-bhan** [comment]
Jesus idk what I did wrong here, gonna have to fix merge conflicts ignore for now

**Comment by luke-bhan** [comment]
@ayazhafiz should be fixed, lmk.

**Comment by ayazhafiz** [comment]
here is my workflow anytime i want to make a change:

```
git checkout master
git fetch origin # or upstream if you forked into your own repo
git rebase origin/master
git checkout -b my_feature_branch
```

Now your branch is updated with changes from master and there will be no merge conflicts because you have completed a rebase.

**Comment by luke-bhan** [comment]
I think i found my issue. For some reason I had all the logo files still added in my git repo on master. So when I would rebase, they would be added back and obviously they aren't on the remote repo. But, thanks this is helpful.

File: libslide/src/bignum/modulo.rs [hunk 1]
```diff
@@ -36,6 +40,9 @@ fn single_skip_mod(lhs: Bignum, rhs: Bignum) -> Bignum {
     if !rhs.dec.is_empty() {
         panic!("The module fuction only supports whole numbers!");
```
Comment (ayazhafiz) [inline libslide/src/bignum/modulo.rs:41]:
```suggestion
        panic!("The modulo fuction only supports whole numbers!");
```

can you fix this real quick

File: libslide/src/bignum/modulo.rs [hunk 2]
```diff
@@ -8,6 +8,10 @@ fn binary_skip_mod(lhs: Bignum, rhs: Bignum) -> Bignum {
...
     }
+    print!("{:?}", rhs.int);
```
Comment (ayazhafiz) [inline libslide/src/bignum/modulo.rs:11]:
```suggestion
```

File: libslide/src/bignum/add.rs [hunk 3]
```diff
@@ -73,6 +74,12 @@ impl ops::Add for Bignum {
         if carry == 1 {
             res_int.push(1);
         }
+        // remove zeros
+        res_int = truncate_zeros(res_int);
+        res_dec = truncate_zeros(res_dec);
+        if res_int.is_empty() && res_dec.is_empty() {
+            is_neg = false;
+        }
```
Comment (ayazhafiz) [inline libslide/src/bignum/add.rs:82]: do we need this?

Comment (luke-bhan) [reply libslide/src/bignum/add.rs:82]: it stops the result of -0. We can do a few things here. We can stop it in the to_string method and allow -0 in our code. Or we can remove it in every method.

Comment (ayazhafiz) [reply libslide/src/bignum/add.rs:82]: let's just handle it in to string

Comment (luke-bhan) [reply libslide/src/bignum/add.rs:82]: ok. will fix.

Comment (timothycobrien) [reply libslide/src/bignum/add.rs]: -0 is also an issue for comparisons, but we can just handle it in there

File: libslide/src/bignum.rs [hunk 4]
```diff
@@ -65,6 +65,9 @@ impl Bignum {
         if !dec.is_empty() && dec.iter().all(|&i| i == 0) {
             dec.clear();
         }
```
Comment (timothycobrien) [inline libslide/src/bignum.rs:67]: Seems like this doesn't remove trailing zeros such as .40000?

File: libslide/src/bignum.rs [hunk 5]
```diff
@@ -65,6 +65,9 @@ impl Bignum {
...
+        if !int.is_empty() && int.iter().all(|&i| i == 0) {
+            int.clear();
+        }
```
Comment (timothycobrien) [inline libslide/src/bignum.rs:70]: Won't remove leading zeros like 00004?

Comment (luke-bhan) [reply libslide/src/bignum.rs:70]: yep good point, will fix;

**Comment by luke-bhan** [comment]
Ok, should be fixed. PTAL @ayazhafiz @timothycobrien

**Comment by timothycobrien** [comment]
Actually one thing I just thought of, do we want to make negative zero a possibility? Or if the int and dec Vecs are both empty, should the is_neg value be set to false as well? Thoughts @ayazhafiz @luke-bhan

No real issue either way? Just need to handle the case in compare `assert(-0==0)`

File: libslide/src/bignum/modulo.rs (hunk 2 above)
Comment (ayazhafiz) [inline libslide/src/bignum/modulo.rs:11]: not removed

**Pull request merged**